lib/SILGen/SILGenExpr.cpp

```diff
@@ -3553,27 +3553,60 @@ RValue RValueEmitter::visitKeyPathExpr(KeyPathExpr *E, SGFContext C) {
   
   auto baseTy = rootTy;
   SmallVector<SILValue, 4> operands;
-  
-  auto lowerSubscriptOperands =
-    [this, &operands, E](const KeyPathExpr::Component &component) {
-      if (!component.getIndexExpr())
-        return;
-      
-      // Evaluate the index arguments.
-      SmallVector<RValue, 2> indexValues;
-      auto indexResult = visit(component.getIndexExpr(), SGFContext());
-      if (isa<TupleType>(indexResult.getType())) {
-        std::move(indexResult).extractElements(indexValues);
-      } else {
+
+  auto lowerSubscriptOperands = [this, &operands,
+                                 E](const KeyPathExpr::Component &component) {
+    if (!component.getIndexExpr())
+      return;
+
+    auto decl = dyn_cast<SubscriptDecl>(component.getDeclRef().getDecl());
+    assert(decl &&
+           "lowerSubscriptOperands must be called with a subscript decl");
+
+    // Evaluate the index arguments.
+    SmallVector<RValue, 2> indexValues;
+    RValue indexResult;
+    if (auto paren = dyn_cast<ParenExpr>(component.getIndexExpr())) {
+      auto param = decl->getIndices()->get(0);
+      if (param->isDefaultArgument())
+        indexResult = visit(param->getDefaultValue(), SGFContext());
+      else
+        indexResult = visit(paren, SGFContext());
+      indexValues.push_back(std::move(indexResult));
+    } else if (auto tupleExpr = dyn_cast<TupleExpr>(component.getIndexExpr())) {
+      for (size_t index = 0; index < tupleExpr->getNumElements(); ++index) {
+        auto param = decl->getIndices()->get(index);
+        if (param->isDefaultArgument())
+          indexResult = visit(param->getDefaultValue(), SGFContext());
+        else
+          indexResult = visit(tupleExpr->getElement(index), SGFContext());
         indexValues.push_back(std::move(indexResult));
       }
+    }
 
-      for (auto &rv : indexValues) {
-        operands.push_back(
-          std::move(rv).forwardAsSingleValue(SGF, E));
-      }
-    };
-  
+    for (auto &rv : indexValues) {
+      operands.push_back(std::move(rv).forwardAsSingleValue(SGF, E));
+    }
+  };
+
+  auto lowerOperands = [this, &operands,
```

Comment:
You should be able to get rid of `lowerOperands`, as only subscript components have index expressions.